staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -416,13 +348,28 @@ func (p *smpPatcher) firstPatchAttempt(currentObject runtime.Object, currentReso
 }
 
 func (p *smpPatcher) subsequentPatchAttempt(currentObject runtime.Object, currentResourceVersion string) (runtime.Object, error) {
+	return subsequentPatchLogic(p.patcher, p, currentObject, currentResourceVersion)
+}
+
+type patchSource interface {
+	originalStrategicMergePatch() (map[string]interface{}, error)
+	computeStrategicMergePatch(unversionedObject runtime.Object, currentVersionedObjMap map[string]interface{}) (map[string]interface{}, error)
```

Comment:
SMP probably isn't the right abstraction here... there are existing bugs about patch retries failing for types that don't support SMP

Reply:
This change is about making the code as it is readable, not about making the code do what we want. Descriptive refactoring, not prescriptive, if you will :)